idseq_dag/steps/run_alignment_remotely.py

```diff
@@ -334,211 +309,78 @@ def run_chunk_wrapper(chunks_in_flight, chunk_output_files, n, mutex, target, ar
-                        # CORRUPT_OUTPUT.
-
-                        # 3. If this is the last attempt, we gotta re-raise the exception.
-
-                        # 4. Elapsed time is only the time spent in alignment.  It excludes the time spent waiting to
-                        # acquire ASGinstance.
-
-                        log.log_event('alignment_remote_error', values={"chunk": chunk_id,
-                                                                        "try_number": try_number,
-                                                                        "CHUNK_MAX_TRIES": CHUNK_MAX_TRIES,
-                                                                        "chunk_status": chunk_status,
-                                                                        "elapsed": elapsed,
-                                                                        "chunk_timeout": chunk_timeout,
-                                                                        "exception": log.parse_exception(e)})
-                        retrying_might_help = chunk_status in (ChunkStatus.CORRUPT_OUTPUT, ChunkStatus.CRASH)
-                        if try_number < CHUNK_MAX_TRIES and retrying_might_help:
-                            # Retry!
-                            continue
-                        else:
-                            # End of the road.
-                            raise
-                    finally:
-                        # None chunk_status indicates code bug above.  An exception has been raised already
-                        # for it, and it says nothing about whether the alignment succeeded or not.
-                        if chunk_status != None:
-                            chunk_status_tracker(service).note_outcome(instance_ip, chunk_id, elapsed, chunk_status, try_number)
-                        self.__delete_remote_dir(remote_work_dir, key_path, remote_username, instance_ip)
-
-            # Upload to s3
-            with self.iostream_upload:  # Limit concurrent uploads so as not to stall the pipeline.
-                command.execute(
-                    command_patterns.SingleCommand(
-                        cmd="aws",
-                        args=[
-                            "s3",
-                            "cp",
-                            "--only-show-errors",
-                            multihit_local_outfile,
-                            os.path.join(self.chunks_result_dir_s3, "")
-                        ]
-                    )
-                )
-            log.write(f"finished alignment for chunk {chunk_id} on {service} server {instance_ip}")
+            log.write(f"finished alignment for chunk {chunk_id} with {self.alignment_algorithm} by lazily fetching last result")
+            return multihit_local_outfile
+
+        # TODO: (tmorse) remove compat hack https://jira.czi.team/browse/IDSEQ-2568
```

Comment:
What is the hack here?
Could `deployment_environment` be null?

Reply:
Yes, as a part of my deployment strategy I am setting this in idseq-web instead of the old method of setting it with the `additional_attributes`. I will ship this first so I want it to be compatible with the old way (null `deployment_environment`) and the new way (null `additional_attributes.environment`). I tested with both

Reply:
Also with the new infra I will set `deployment_environment` in the job definition of the job running the dag itself so this doesn't need to be set explicitly.